src/rabbit_auth_backend_oauth2.erl

```diff
@@ -140,7 +144,7 @@ validate_token_expiry(#{}) -> ok.
 check_token(Token) ->
     case uaa_jwt:decode_and_verify(Token) of
         {error, Reason} -> {refused, {error, Reason}};
-        {true, Payload} -> validate_payload(post_process_payload(Payload));
+        {true, Payload} -> post_process_identityserver_roles(validate_payload(post_process_payload(Payload)));
```

Comment:
@michaelklishin I really dont like this line because its break Single responsibility principle. Method check_token() shouldnt modify Payload content. Anyway, Initialy i wanted to extendt post_process_payload() method with "picklock" functionaliy, but its much easier to works on data returned by validate_payload() method - where "resource_server_id" is removed from scope prefix. IMO, this piece of code requires some refactor.

Reply:
I agree `check_token/1` could be renamed. Or the post-processing of the payload could happen after the call to it (it's only called in a couple of places). We can refactor that later.

Are you sure there's no way to handle the IdentifyServer case in the `post_process_payload/1`, just like for Keycloak? This would make this more consistent.

Reply:
We can rename the function, it's a module-local implementation detail. We can also transform the resulting JWT map in a separate function.

FTR, the `check_token/1` part is now only called at authentication time.

Reply:
FTR, what does it mean?

Reply:
For the record.